main.js

```diff
@@ -1,20 +1,63 @@
 import repos from "/data/repos.js";
+import { searchbarMap, tableMap, paginationMap } from "/data/html-map.js";
+import { htmlByMap, prepareReposForView, updateTable, filter } from "/data/function.js";
 
 export default function buildPage() {
-    const root = document.querySelector("div#root");
-    root.innerText = "Hello!";
+  const root = document.querySelector("div#root");
+
+  // build page start ---------------------------------
+  htmlByMap(root, searchbarMap);
+
+  var pagesNumber =
+    document.getElementById("per-page").value !== ""
+      ? repos.length / document.getElementById("per-page").value
+      : 1;
+
+  paginationMap.children = paginationMap.generatePages(50);
```

Comment:
good idea overall with html-map component, 
which makes UI configurable

i guess project still requires some improvements in various places.

major potential problem with your design is that you apply [Builder](https://refactoring.guru/design-patterns/builder) and [Composite](https://refactoring.guru/design-patterns/composite) design patterns in the place where [Decorator](https://refactoring.guru/design-patterns/decorator) could work better.  Search, Sort and Pagination functionality are just "Decorations".
Options that user chose to apply for these decorations are global config parameters and could be also names as "state variables".